azurerm/internal/services/trafficmanager/resource_arm_traffic_manager_profile.go

```diff
@@ -106,6 +106,25 @@ func resourceArmTrafficManagerProfile() *schema.Resource {
 							},
 						},
 
+						"custom_headers": {
+							Type:     schema.TypeList,
```

Comment:
As this is a map of name->values should we just use a type map here intead?

Reply:
That was my initial thought, since that's how JS does it with the [Headers object constructor][mdn-headers], but

1. a request is technically [allowed to have multiple headers with the same name][stackoverflow-multiple-headers] -- although it's equivalent to a comma-separated list, it can sometimes be nicer (e.g. with long entries) to have separate headers
2. the Azure REST interface uses a list of key-value pairs, and my impression of this codebase was that it tries to cleave closely to Azure where there isn't an opportunity for major simplification.


[mdn-headers]:https://developer.mozilla.org/en-US/docs/Web/API/Headers/Headers
[stackoverflow-multiple-headers]:https://stackoverflow.com/questions/4371328/are-duplicate-http-response-headers-acceptable

Reply:
Actually, just realized `azurerm_traffic_manager_endpoint` already implements `custom_header`. Now that I've fixed pluralization, this implementation looks consistent with that one, and both seem consistent with their respective Azure REST equivalents :slightly_smiling_face: